### PR TITLE
Inject eeg and marker sources into EEG_data

### PR DESCRIPTION
**Pull request by gregbci**
I ended up knocking off all the tasks related to pulling the LSL and XDF dependencies out of EEG_data.

[B4K-225](https://bci4kids.atlassian.net/browse/B4K-225)
[B4K-226](https://bci4kids.atlassian.net/browse/B4K-226)
[B4K-235](https://bci4kids.atlassian.net/browse/B4K-235)

Key changes:

* The init has changed to allow injection of sources: EEG_data(classifier, eeg_source, marker_source,subset).  The init performs some tweaks to the incoming EEG data re: DSI7, DSI42 and trigger channels.
* Two abstract "source" types have been created: EegSource and MarkerSource
* I added LSL and XDF versions of these sources and updated tests, examples accordingly
* load_offline_eeg_data(), stream_online_eeg_data() and their helper functions no longer exist (replaced by sources)
* I added tests for the sources and a little bit of EEG_data to demonstrate injecting a mock source object for tests (probably better tools out there for this, but same idea)

### CODE REVIEW

File: bci_essentials/eeg_data.py [hunk 1]
```diff
@@ -1286,7 +874,7 @@ def main(
                         self.marker_data[self.marker_count][0]
                         == "Done with all RS collection"
                     ):
-                        self.package_resting_state_data()
+                        self.__package_resting_state_data()
                         self.marker_count += 1
 
                     elif self.marker_data[self.marker_count][0] == "Trial Started":
```
Comment (brianjohannes) [inline bci_essentials/eeg_data.py:880]: Minus 400 lines!! Nice!

**Review by brianjohannes — APPROVED** [review]
Hey Greg, looks great, the only issues I found were little typos. Very excited to see EEG_data getting slimmed down.

**Comment by gregbci** [comment]
@brianjohannes I can make those typo fixes before you merge

**Comment by gregbci** [comment]
Done

**Pull request merged**